sdlib/d/gc/tcache.d

```diff
@@ -151,7 +151,15 @@ public:
 	}
 
 	void destroyThread() {
-		free(tlsSegments.ptr);
+		// The suspend hook may also try to free. Avoid the race
+		// condition.
+		state.enterBusyState();
+		scope(exit) state.exitBusyState();
+		if (tlsSegments.ptr !is null) {
+			free(tlsSegments.ptr);
+			tlsSegments = [];
+		}
```

Comment:
Can we extract that into its own function and call it from here and from the dmd hook?

Reply:
Done